### PR TITLE
Ssdeep similarity

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39331

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by JasBeilin** [comment]
After consulting with @glicht we decided to use cli ssdeep as the python pack uses a license which we cannot use.

File: Packs/CommonScripts/ReleaseNotes/1_4_15.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### New: SSDeepSimilarity
+- This script finds similar files that can be related to each other by fuzzy hash (SSDeep) . (Available from Cortex XSOAR 5.5.0).
```
Comment (bakatzir) [inline Packs/CommonScripts/ReleaseNotes/1_4_15.md:4]:
```suggestion
- This script finds similar files that can be related to each other by fuzzy hash (SSDeep). (Available from Cortex XSOAR 5.5.0).
```

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| ssdeep_hash | The SSDeep hash to check for similarity against. |
+| ssdeep_hashes_to_compare | A list of SSDeep hashes to check for similarity to the ssdeep_hash input. |
+| output_key | The context key to which the list of SSDeep hashes will be outputted. |
+
+## Outputs
+---
+There are no outputs for this script.
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/README.md:23]: please add here the SourceHash, compared_hashes with a placeholder to the output key

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+commonfields:
+  id: SSDeepSimilarity
+  version: -1
+name: SSDeepSimilarity
+script: ''
+type: python
+tags: []
+comment: This script finds similar files that can be related to each other by fuzzy
+  hash (SSDeep) .
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.yml:9]:
```suggestion
comment: This script finds similar files that can be related to each other by fuzzy
  hash (SSDeep).
```

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  description: The context key to which the list of SSDeep hashes will be outputted.
+  defaultValue: SSDeepSimilarity
+outputs:
+  - contextPath: SSDeepSimilarity.compared_hashes.similarityValue
+    description: The difference calculation score between the ssdeep_hash and the compared
+      hash.
+    type: number
+  - contextPath: SSDeepSimilarity.compared_hashes.hash
+    description: The hash compared to the ssdeep_hash.
+    type: string
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.yml:30]: so do we or do we not have OOTB outputs?

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity_test.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+INPUT_INVALID_CASES = [
+    (CASE_ANCHOR_EMPTY, 'Please provide an hash to compare to.'),
+    (CASE_HASH_LIST_EMPTY, 'Please provide at least one hash to compare with.'),
+
+]
+
+
+@pytest.mark.parametrize('case_inputs, expected_error_msg', INPUT_INVALID_CASES)
+def test_handle_inputs_fails_on_invalid_input(case_inputs, expected_error_msg):
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity_test.py:21]: let's add a case for giving a SSDeep hash that is invalid(not valid chars inside/length/etc)

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+    for current_hash in hashes_to_compare:
+        hashes_list_to_file += f'{current_hash},"{current_hash}"\n'
+
+    res = run_ssdeep_command(anchor_hash_to_file, hashes_list_to_file)
+    hashes_outputs = _format_results(res)
+
+    md = tableToMarkdown(anchor_hash, hashes_outputs)
+    return CommandResults(
+        outputs_prefix=output_key,
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py:59]: let's add the `SourceHash` as a key to merge on.

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        formatted_res.append({
+            'hash': result[0].strip('"'),
+            'similarityValue': int(result[2])
+        })
+    return formatted_res
+
+
+def run_ssdeep_command(anchor_hash: str, hashes_to_compare: str):
+    with tempfile.NamedTemporaryFile() as hash1:
+        with tempfile.NamedTemporaryFile() as hash2:
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py:38]: let's give more meaningful var names here

File: Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    for result in results:
+        result = result.split(',')
+        formatted_res.append({
+            'hash': result[0].strip('"'),
+            'similarityValue': int(result[2])
+        })
+    return formatted_res
+
+
+def run_ssdeep_command(anchor_hash: str, hashes_to_compare: str):
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/SSDeepSimilarity/SSDeepSimilarity.py:36]:
```suggestion
def run_ssdeep_command(anchor_hash: str, hashes_to_compare: str) -> list:
```

**Review by bakatzir — COMMENTED** [review]
love the coverage here.

**Review by bakatzir — APPROVED** [review]

**Pull request merged**